docs/readmes/proposals/p022_enodebd_enhancements.md

```diff
@@ -0,0 +1,177 @@
+
+**2.2 Plug-n-Play**
+The radio onboarding process for Magma currently follows a pre-provisioning workflow. A user with login credentials for the NMS must complete the following steps:
+1. Add a new radio element to the NMS with the following fields:
+- Name
+- Serial number
+- Description
+- Externally managed
+- Device Class 
+- Bandwidth
+- Cell ID
+- RAN Config (FDD/TDD)
+- EARFCNDL
+- Special Subframe Pattern
+- Subframe Assignment
+- PCI
+- TAC
+- Transmit Enable
+
+2. Assign the radio to an AGW
+
+After this process has been completed the enodebd process in the AGW will now respond to the initial TR-069 Inform message and attempt to configure the radio. Enodebd uses the serial number as the key for matching informs to known radio elements. 
+
+If an inform message arrives at enodebd with a serial number that is not known, the TR-069 session is terminated and no further action is taken. 
+
+**2.3 Firmware Update**
+Magma TR-069 ACS functionality in enodebd currently does not support firmware update procedures. 
+
+### **3 Implementation**
+
+**3.1 Plug-n-Play**
+
+**3.1.1 Block Diagram**
+![block diagram](https://user-images.githubusercontent.com/93994458/141427317-8e19f9b3-b789-4b8b-9117-c2a4024b9089.png)
+
+**3.1.2 Call Flow**
+![onboarding flow](https://user-images.githubusercontent.com/93994458/141427664-16110fc4-61a4-4bfa-a7b0-b3f379c62150.png)
+
+**3.1.3 PnP Scope of Change**
+
+**3.1.3.1 eNodeBD Service (Modified)**
+The enodebd service will be updated to support 2 modes of operation, auto-discovery enabled/disabled. 
+_Auto-Discovery Disabled_
+When Auto-Discovery is disabled enodebd will operate as it does today. When a TR-069 inform message is received with a serial number unknown to enodebd, it will terminate the session without further action. 
+_Auto-Discovery Enabled_
+When Auto-discovery is enabled, enodebd will receive a new inform, check it’s local configuration to see if the serial number is known, if it is not, it will forward the serial number and device ID field contents to Control Proxy with the endpoint specified as Onboarding Service. 
+
+**3.1.3.2 Control Proxy Service (Modified)**
+The Control Proxy service will be modified to add Onboarding Service onboardd to the service registry. This new endpoint will be used for transporting messages between enodebd and onboardd.
+
+**3.1.3.3 Onboardd Service (New)**
+The Onboarding Service onboardd will be a new service added to support automated radio onboarding. The primary function will be to respond to enodebd new radio messages and using the API, create a new radio element and assign it to the source AGW. 
+
+In addition, when the new eNB is a CBRS CBSD, onboardd will also create a new radio element in the Domain Proxy (DP). When the radio is a Cat A indoor with a GPS derived location, the onboardd configuration will be sufficient for the DP to register and obtain a grant for the new eNB. 
+
```

Comment:
Let's also add a description here to note that the `onboardd` service and plug and play feature will only take into account `Managed` eNBs due to the TR-069 session.

```suggestion
The onboardd service will only support magma Managed eNodebs, this is due to the requirement of establishing a TR-069 server session to further configure the radio device. Externally managed devices do not setup a TR-069 session.
```